docs/testing/testcafe_functional_testing.md

```diff
@@ -0,0 +1,150 @@
+---
+
+## TestCafe: Functional tests
+
+Templated sample TestCafe automation framework for PROJECT_NAME, bootstrapped using `@amidostacks/scaffolding-cli`.
+
+## Getting started
+
+Ensure that [Node.js](https://nodejs.org/) and [npm](https://www.npmjs.com/) are installed with the versions specified in [package.json](./package.json).
+
+### Dependencies
+
+```bash
+npm install
+```
+
+### End to End (E2E) Tests
+
+```bash
+npm run test:e2e
+```
+
+This should launch [TestCafe](https://devexpress.github.io/testcafe/documentation/getting-started/) and run the tests in the browsers specified in [.testcaferc.json](./.testcaferc.json) in the headless mode.
+
+### Accessibility Tests with aXe
+
+```bash
+npm run test:axe
+```
+
+This will launch [TestCafe](https://devexpress.github.io/testcafe/documentation/getting-started/) and run the tests in the browsers specified in [.testcaferc.json](./.testcaferc.json) in headless mode.
+
+This will validated using axe-core, from the Deque family of products: [aXe](https://www.deque.com/axe/). When developing, we expect to support WCAG 2.1 Level AA ["wcag21aa"] at a minimum. The additional configuration can be changed by defining the [context and options](https://www.npmjs.com/package/axe-testcafe#axe-options).
+
+## Browser configuration
+
+We assume that most users will have Chrome installed on their operating system. If this is not the case, the base configuration file  
+  "browsers": [
+    "chrome:headless",
+    "firefox:headless"
+  ],
+
+To bypass the browser config in the [.testcaferc.json](./.testcaferc.json) config file, simply run passing in the browser arg:
+
+```bash
+npm run test:e2e -- chrome
+```
+
+See [browsers](https://devexpress.github.io/testcafe/documentation/using-testcafe/common-concepts/browsers/) for more information.
+
+## Environment variables
+
+### E2E Tests
+
+We can use the [.env](./.env) file to specify defaults instead of exporting locally, or using your IDE.
+
+_⚠️ IMPORTANT: if you have defined an environment variable, then `dotenv` will not override this with the values in the `.env` file. The environment should always hold the source of truth._
+
+- Base URL: `process.env.APP_BASE_URL` (defaults to deployed dev environment `https://dev-netcore-app.nonprod.amidostacks.com `)
+- Base URL: `process.env.APP_BASE_PATH` (optional, default to deployed dev environment `/web/stacks`)
+- Base URL API: `process.env.MENU_API_URL` (defaults to deployed dev environment `https://dev.amidostacks.com/api/menu`)
+- NODE_ENV: `process.env.NODE_ENV` (should be `production` for a deployed runnable webapp)
+
+An example of your environment variable configuration for running against a locally hosted server is as follows.
+
+For Linux/Mac (replacing `export` with `set` for Windows):
+
+``` bash
+export NODE_ENV=development 
+export PORT=3000
+export APP_BASE_URL=APP_BASE_URL_VALUE
+export MENU_API_URL=MENU_API_URL_VALUE
+export APP_BASE_PATH=""
+```
+
+## Running tests in Docker
+
```

Comment:
Given the instructions here there is a lot of dependencies to get this to run.  I would add in links to docker and how to set this up here otherwise you will have no context